runelite-client/src/main/java/net/runelite/client/plugins/playerindicators/PlayerIndicatorsTileOverlay.java

```diff
@@ -62,7 +65,18 @@ public Dimension render(Graphics2D graphics)
 		{
 			final Polygon poly = player.getCanvasTilePoly();
 
-			if (poly != null)
+			if (client.isFriended(player.getName(), false) && config.drawFriendsTiles() && poly != null && config.highlightOfflineFriends() && config.highlightFriends())
+			{
+				if (config.disableFriendHighlightIfClanMember() && !player.isClanMember())
+				{
+					OverlayUtil.renderPolygon(graphics, poly, color);
+				}
+				else if (!config.disableFriendHighlightIfClanMember())
+				{
+					OverlayUtil.renderPolygon(graphics, poly, color);
+				}
+			}
+			if (config.drawTiles() && poly != null)
 			{
 				OverlayUtil.renderPolygon(graphics, poly, color);
 			}
```

Comment:
This can still be called even when the previous if-block is true and renderPolygon() is already called once. Which won't be efficient performance wise.